File: 002.c
#include <stdio.h>

int
main (void) {
    int total = 0;
    int last = 1;
    int current = 2;

    while (current < 4000000) {
        if (current % 2 == 0) {
            total += current;   
        }
    int temp = last;
    last = current;
    current = last + temp;
    }

    printf("%d\n", total);
    return 0;
}

File: 002.rs
fn main() {
    let mut previous = 1;
    let mut current = 2;

    let mut sum = 0;

    while current < 4000000 {
        if current % 2 == 0 {
            sum += current;
        }
        let temp = current;
        current += previous;
        previous = temp;
    }
    println!("{}", sum);
}
